### PR TITLE
Tunnels and Coins and Guards

### PR DESCRIPTION
**Pull request by ms2300**
- Fixes coins spawning outside stadium
- Fixes streaker running outside grass
- Adds randomly spawning tunnels (not fully implemented)
- Adds a guard class with implementation needed later from me
- Refactors the static constants class
- Adds a paths class
- Small amounts of cleanup in random classes

### CODE REVIEW

File: src/StreakerGame.java [hunk 1]
```diff
@@ -45,22 +44,21 @@ public static void main(String[] args) {
...
-                double t = nanot / constants.getPrecision();
+                double t = nanot / Constants._PRECISION;
                 background.loop();
                 background.setBackgroundSpeed(gc);
                 character.handleVelocity(input);
                 character.handleCharacterPosition();
                 character.updateVelocity(elapsedTime);
                 character.render(gc, t);
                 for (Coin coin : coins) {
-                    if (coin.getY() > constants.getScreenHeight()) {
```
Comment (kramsey458) [inline src/StreakerGame.java]: Lines 60-69 got moved into a method called handleCoinIntersects. Should not change anything you did.

File: src/StreakerGame.java [hunk 2]
```diff
@@ -105,29 +109,36 @@ private void setupGameState(Stage stage) {
...
         graphicsController = new GraphicsController(gc);
+        character = new Streaker();
         setOnKeyPress();
         setOnKeyRelease();
         createCoins();
+        createTunnels();
         setInitialScore();
     }
 
     private void createCoins() {
```
Comment (kramsey458) [inline src/StreakerGame.java]: Lines 126-133 were moved into a seperate method as well in the coins class.

**Review by kramsey458 — DISMISSED** [review]
Looks good. Ill redo some main loop refactoring after these changes.

**Comment by kramsey458** [comment]
Change "Revert Issue #32" in above comment to "Revert Pull Request 42"

**Comment by ms2300** [comment]
This pull request should be good to go, merging problems are dons

**Comment by carolinecullen** [comment]
wow, way to go heres a cookie

**Pull request merged**